### PR TITLE
implemented the join program for the template page and made the chang…

### PR DESCRIPTION
**Pull request by farhanp1502**
…es for the template for the theme and api services

### CODE REVIEW

File: projects/project-player-library/src/lib/pages/preview-details-page/preview-details-page.component.ts [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
      this.getProjectTemplate()
-      this.getData(this.solutionId)
+      this.setActionsList();
+      this.initializeTasks()
     })
   }
   ngOnInit(): void {
   }
 
+  ngOnChanges(changes: SimpleChanges) {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/preview-details-page/preview-details-page.component.ts:33]: Why is this function added here?

File: projects/project-player-library/src/lib/shared/start-improvement-popup/start-improvement-popup.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="p-5">
+    <div class="pb-4 text-2xl font-bold">
+        {{ "START_IMPROVEMENT" | translate}} ?
+    </div>
+    <div class="pb-4 font-bold">{{"START_IMPROVEMENT_MSG" | translate }}</div>
+    <div class="pb-4">{{ "START_IMPROVEMENT_MSG_CONFIRM" | translate }}</div>
+    <div class="flex justify-center">
+        <button class="py-3 px-1 primary-button" mat-flat-button (click)="closePopup(true)">{{"JOIN_PROGRAM"| translate}}</button>
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/start-improvement-popup/start-improvement-popup.component.html:8]: Chnage the button label with the Start improvement

File: projects/project-player-library/src/lib/shared/task-card/task-card.component.ts [hunk 3]
```diff
@@ -29,5 +34,25 @@ export class TaskCardComponent {
     if (!this.submittedImprovement && !this.startImprovement) {
       this.routerService.navigate(`/task-details/${data}/${this.projectId}`);
     }
+    else if(!this.submittedImprovement && this.startImprovement){
+      this.showConfirmJoinProgram()
+    }
+  }
+  showConfirmJoinProgram() {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/task-card/task-card.component.ts:41]: Change to function name

File: projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.html [hunk 4]
```diff
@@ -3,53 +3,46 @@
...
-                <div>{{ "NO_FILES" | translate:{ fileType: 'Image' } }}</div>
+    <mat-tab [label]="'IMAGES' | translate">
+        <div class="p-4 w-full text-center" *ngIf="!isImages?.project && !isImages?.task">
+            <div>{{ "NO_FILES" | translate:{ fileType: 'Image' } }}</div>
         </div>
-        <div class="image-tasklist p-4" >
-            <div class="flex items-center"   *ngIf="isImages?.project ">
-                <div class="w-auto mr-3" >{{"LEVEL_EVIDENCE" | translate:{type:'Project'} }}</div>
-                <span class="h-[4px]  flex-grow bg-[var(--primary-color)]"></span>
+        <div class="image-tasklist p-4">
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.html:10]: Revert these formatting changes

File: projects/project-player-library/src/lib/pages/preview-details-page/preview-details-page.component.ts (hunk 1 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/preview-details-page/preview-details-page.component.ts:33]: Removed

File: projects/project-player-library/src/lib/shared/start-improvement-popup/start-improvement-popup.component.html (hunk 2 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/start-improvement-popup/start-improvement-popup.component.html:8]: done

**Pull request merged**